### ISSUE TITLE
Expose additional optional parameters in SLURM connector (e.g., number of nodes requested)

### ISSUE THREAD

**Issue by paoloviviani** [alpha-unito/streamflow#14]
It would be useful to expose (some of) the parameters of the SLURM job in the deployment section of streamflow.yml, as in the following example:
```
deployments:
  slurm_cluter:
    type: slurm
    config:
      file: environments/sbatch.sh
      nodes: 10
      gres: "gpu:1"
      sbatch_options: "--exclusive --reservation=resv_id --time=00:01:00"
```
and then reflect them in the sbatch.sh file as jinja templates as
```
#SBATCH --ntasks-per-node={{streamflow_slurm_nodes}}
#SBATCH --gres={{streamflow_slurm_gres}}
#SBATCH {{streamflow_sbatch_options}}
```
so that the user that would like to change them does not have to dig into the batch script and keep track of the content of multiple files, but the information is reported at the top level and propagated below.

Exactly which SLURM parameters should be exposed can is open to discussion (I would say number of nodes and GPUs as the baseline), then we may dump everything else in a `sbatch_options` string paramter.

**Comment by paoloviviani** [comment]
After some discussion we revised the idea
streamflow.yml
```yaml 
deployments:
  slurm_cluter:
    type: slurm
    config:
      file: environments/sbatch.sh
      options:
        nodes: 10
        gres: "gpu:1"
        time: 01:00:00
```
template_batch_script.sh
```
#SBATCH --{{key_1}}={{value1}}
#SBATCH --{{key_2}}={{value2}}
#SBATCH --{{key_3}}={{value3}}
```

**Comment by GlassOfWhiskey** [comment]
Solved by #151 and #227.
